source/base/parameter_handler.cc

```diff
@@ -1164,12 +1164,11 @@ ParameterHandler::recursively_print_parameters(
                            p->second.get<std::string>("value").length());
               }
 
-          // print entries one by one. make sure they are sorted by using
-          // the appropriate iterators
+          // print entries one by one
           bool first_entry = true;
-          for (boost::property_tree::ptree::const_assoc_iterator p =
-                 current_section.ordered_begin();
-               p != current_section.not_found();
+          for (boost::property_tree::ptree::const_iterator p =
+                 current_section.begin();
+               p != current_section.end();
```

Comment:
Can't you use `range-based` for loops in all these places?

Reply:
You are absolutely right, I've updated the modifications!